### PR TITLE
TEST: fixed valgrind args

### PR DESCRIPTION
**Pull request by hoopoepg**
fixes https://github.com/openucx/ucx/issues/2762

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7374/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@hoopoepg seems it's working because now jenkins detected all the leaks :)

**Comment by hoopoepg** [comment]
yep :)
it seems there is destructor of tests is not called and valgrind reports a tons of leaks on tests constructed... looking over it

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7376/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7379/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7381/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
@yosefe will it work?

**Comment by hoopoepg** [comment]
@yosefe there are more PR's which may break valgrind :)

File: src/ucm/malloc/malloc_hook.c [hunk 1]
```diff
@@ -581,20 +619,25 @@ static int ucm_malloc_mallopt(int param_number, int value)
...
-    { "calloc", ucm_calloc },
-    { "valloc", ucm_valloc },
-    { "posix_memalign", ucm_posix_memalign },
-    { "setenv", ucm_setenv },
-    { UCM_OPERATOR_NEW_SYMBOL, ucm_operator_new },
-    { UCM_OPERATOR_DELETE_SYMBOL, ucm_operator_delete },
-    { UCM_OPERATOR_VEC_NEW_SYMBOL, ucm_operator_vec_new },
-    { UCM_OPERATOR_VEC_DELETE_SYMBOL, ucm_operator_vec_delete },
-    { NULL, NULL }
+    { .symbol = "free", .value = ucm_free, .black_list = ucm_malloc_black_list },
```
Comment (yosefe) [inline src/ucm/malloc/malloc_hook.c]: maybe we should use a global blacklist?

Comment (hoopoepg) [reply src/ucm/malloc/malloc_hook.c]: there are set of patches specific for CUDA only - it should not be blacklisted for CUDA modules

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7384/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
@yosefe anything else to fix?

**Pull request merged**